lib/services/record_position_updater.js

```diff
@@ -19,6 +26,11 @@ class RecordOrderUpdater {
     return this.section.getRecords({ order: [['position', 'ASC']] });
   }
 
+  async _append() {
+    const maxPosition = Utils.chain(this.siblings).map(s => s.position).max().value();
```

Comment:
I think there's still a shlemiel the painter problem here where inserts will take linearly longer based on the length of the records in the section, however, it might not even be noticeable until you get to millions of records so *shrug*. One potential optimization for this solution might be something like `select max(position) from records where section_id = n;` and adding 1 instead of selecting all records in the section and then mapping for max position.

Reply:
Good point. I just pushed a change to use `Record.max`.